sdks/python/apache_beam/coders/typecoders.py

```diff
@@ -154,4 +157,25 @@ def verify_deterministic(self, key_coder, op_name, silent=True):
     else:
       return key_coder
 
+
+class FirstOf(object):
```

Comment:
(perhaps we want a CoderProvider class that is just the interface for from_type_hint, which relevant Coders can implement as well; no action needed)